### ISSUE TITLE
Using Kitty as a custom terminal for Alfred

### ISSUE THREAD

**Issue by pyrho** [pyrho/25.wf-comments#2]

**Comment by mcclowes** [comment]
This works if kitty is already open, but doesn't if it's not - any idea how to get it to open kitty if it's not open?

**Comment by pyrho** [comment]
Indeed, I mostly have kitty launched at startup so I didn't see the issue.
I updated the script on the blog post, basically just put the `tell application Kitty to activate` line first, see [here](https://git.sr.ht/~pyrho/blog/commit/64859d6674898bb89a5e68ad4644c798842efebe) for a full diff.

**Comment by mcclowes** [comment]
Makes sense, thanks man! 🐱

**Comment by camflint** [comment]
Thanks for this! It works perfectly. 🍻

**Comment by pyrho** [comment]
Happy to hear that ! 🍻

**Comment by Liebhecht** [comment]
Sadly this is not working on Alfred 5. It opens kitty but doesn't input the text from Alfred.

**Comment by 3-io** [comment]
@Liebhecht I've found the solution to this issue, you need to manually create the socket file.

```py
python -c "import socket as s; sock = s.socket(s.AF_UNIX); sock.bind('/tmp/mykitty')"
```

Once created, restart Kitty and run a command from Alfred; the first time you do this, it will request permissions to access System Events. Grant them and you should be all set.

Confirmed working with:
- MacOS Ventura 13.1
- Kitty 0.25.0
- Alfred 5.1 [2134]

**Comment by pyrho** [comment]
Hey @djames-bloom thanks for sharing! I guess a downside of that approach is that you would have to create this socket again each time you reboot.

I looked a bit into it and came up with this apple script, which should Just Work ™️  :

```applescript
on alfred_script(q)
    tell application "kitty" to activate
    tell application "System Events"
        repeat until (exists file "/tmp/kittynyx.sock")
            delay 6
        end repeat
    end tell
    do shell script "/Applications/kitty.app/Contents/MacOS/kitty @ --to unix:/tmp/kittynyx.sock new-window --new-tab"
    tell application "System Events" to keystroke q
    tell application "System Events"
        key code 36 -- enter key
    end tell
end alfred_script
```


The part I added is: 
```applescript
    tell application "System Events"
        repeat until (exists file "/tmp/kittynyx.sock")
            delay 6
        end repeat
    end tell
```

This tells the script to wait until kitty has actually created the socket before trying to write to it.

*edit*: You need to rename `/tmp/kittynyx.sock` to the name of **your** socket.  
And ⚠️ and this code has no safeguards, so if, for some reason, Kitty fails to create the socket upon activation this will lead to an infinite loop, not sure if Alfred would kill this after a given timeout or not.

**Comment by Cliffback** [comment]
Great work, thanks. I did some modifications to the script myself, so that it pastes the command into the current open window by default, and if "-n" is written first after >, then it will open it in a new window in stead (as in this script)

Put it up here if anyone is looking for something similar [alfred-kitty-integration](https://github.com/Cliffback/alfred-kitty-integration)

**Comment by pyrho** [comment]
Awesome, thank you for sharing !
